tests/src/drivers/test_driver_size.c

```diff
@@ -25,73 +25,34 @@
 #include "test/drivers/size.h"
 #include "psa/crypto.h"
 
-typedef struct {
-    unsigned int context;
-} test_driver_key_context_t;
-
-/*
- * This macro returns the base size for the key context. It is the size of the
- * driver specific information stored in each key context.
- */
-#define TEST_DRIVER_KEY_CONTEXT_BASE_SIZE sizeof( test_driver_key_context_t )
-
-/*
- * Number of bytes included in every key context for a key pair.
- *
- * This pair size is for an ECC 256-bit private/public key pair.
- * Based on this value, the size of the private key can be derived by
- * subtracting the public key size below from this one.
- */
-#define TEST_DRIVER_KEY_CONTEXT_KEY_PAIR_SIZE      65
-
 /*
- * Number of bytes included in every key context for a public key.
- *
- * For ECC public keys, it needs 257 bits so 33 bytes.
+ * This macro returns the base size for the key context when SE does not support storage.
+ * It is the size of the metadata that gets added to the wrapped key.
+ * In its test functionality the metadata is just some padded prefixing to the key.
  */
-#define TEST_DRIVER_KEY_CONTEXT_PUBLIC_KEY_SIZE    33
+#define TEST_DRIVER_KEY_CONTEXT_BASE_SIZE  PSA_CRYPTO_TEST_DRIVER_OPAQUE_PAD_PREFIX_SIZE
 
-/*
- * Every key context for a symmetric key includes this many times the key size.
- */
-#define TEST_DRIVER_KEY_CONTEXT_SYMMETRIC_FACTOR   0
 
-/*
- * If this is true for a key pair, the key context includes space for the public key.
- * If this is false, no additional space is added for the public key.
- *
- * For this instance, store the public key with the private one.
- */
-#define TEST_DRIVER_KEY_CONTEXT_STORE_PUBLIC_KEY   1
-
-size_t mbedtls_test_size_function(
+size_t mbedtls_test_opaque_size_function(
     const psa_key_type_t key_type,
     const size_t key_bits )
 {
     size_t key_buffer_size = 0;
 
-    if( PSA_KEY_TYPE_IS_KEY_PAIR( key_type ) )
+    if( key_bits != 0 )
     {
-        int public_key_overhead =
-            ( ( TEST_DRIVER_KEY_CONTEXT_STORE_PUBLIC_KEY == 1 )
-              ? PSA_EXPORT_KEY_OUTPUT_SIZE( key_type, key_bits ) : 0 );
-        key_buffer_size = TEST_DRIVER_KEY_CONTEXT_BASE_SIZE +
-                          TEST_DRIVER_KEY_CONTEXT_PUBLIC_KEY_SIZE +
-                          public_key_overhead;
+        key_buffer_size = PSA_EXPORT_KEY_OUTPUT_SIZE( key_type, key_bits );
+        if( key_buffer_size == 0 )
+            return( key_buffer_size );
+        /* Include spacing for base size overhead over the key size
+         * */
+        key_buffer_size += TEST_DRIVER_KEY_CONTEXT_BASE_SIZE;
     }
```

Comment:
@ronald-cron-arm , this is the sizing am referring to.

Reply:
This is test code thus we can do whatever we want here. If we decide that the size of key contexts for the opaque test drivers is PSA_EXPORT_KEY_OUTPUT_SIZE plus TEST_DRIVER_KEY_CONTEXT_BASE_SIZE then so be it. I don't see any problem here.